crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -890,6 +890,75 @@ impl<'w> EntityWorldMut<'w> {
         self
     }
 
+    /// Removes a dynamic [`Component`] from the entity.
+    ///
+    /// You should prefer to use the typed API [`EntityWorldMut::remove`] where possible.
+    ///
+    /// # Safety
+    ///
+    /// - [`ComponentId`] must be from the same world as [`EntityWorldMut`]
+    pub unsafe fn remove_by_id(&mut self, component_id: ComponentId) -> &mut Self {
+        let archetypes = &mut self.world.archetypes;
+        let storages = &mut self.world.storages;
+        let components = &mut self.world.components;
+        let entities = &mut self.world.entities;
+        let removed_components = &mut self.world.removed_components;
+
+        let (bundle_info, _storage_type) = self
+            .world
+            .bundles
+            .init_component_info(components, component_id);
+
+        let old_location = self.location;
+        // SAFETY: `archetype_id` exists because it is referenced in the old `EntityLocation` which is valid,
+        // components exist in `bundle_info` because `Bundles::init_info` initializes a `BundleInfo` containing all components of the bundle type `T`
+        let new_archetype_id = remove_bundle_from_archetype(
```

Comment:
Can we make this safe? `Bundles::init_component_info` (and not `Bundles::init_info`, like the safety comment claims) panics if no component with id `component_id` exists, so `remove_bundle_from_archetype` should always be sound to call.